virtualmachine/vsphere/util.go

```diff
@@ -1369,6 +1411,23 @@ var getVMLocation = func(vm *VM, dcMo *mo.Datacenter) (l location, err error) {
 		}
 		l.ResourcePool = *crMo.ResourcePool
 		l.Networks = crMo.Network
+	case DestinationTypeResourcePool:
+		var rp *mo.ResourcePool
+		dc := object.NewDatacenter(vm.client.Client, dcMo.Self)
+		// Set datacenter
+		vm.finder.SetDatacenter(dc)
+
+		rp, err = findResourcePoolByMOID(vm, vm.Destination.MOID)
```

Comment:
Does this mean that MOID is mandatory if resource pool has to be used? Is it not possible to find resource pool by "Name" as we do in case of Cluster? It would be better if we can support both.

Reply:
Multiple resource pools and child resource pools can have same name, so we right now only support resource pool with MOID.
It will be ambiguous if we support resource pool with name.